airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -122,6 +125,44 @@ def ti_update_state(
         )
     elif isinstance(ti_patch_payload, TITerminalStatePayload):
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
+    elif isinstance(ti_patch_payload, TIDeferredStatePayload):
+        trigger_row = Trigger(
+            classpath=ti_patch_payload.classpath,
+            kwargs=ti_patch_payload.kwargs,
+            created_date=ti_patch_payload.created_date,
+        )
+        session.add(trigger_row)
+        session.flush()
+
+        ti = session.query(TI).filter(TI.id == ti_id_str).one_or_none()
+
+        # Calculate timeout too if it was passed
+        trigger_timeout: datetime | None = None
+        if ti_patch_payload.timeout is not None:
+            trigger_timeout = timezone.utcnow() + ti_patch_payload.timeout
+        else:
+            trigger_timeout = None
+
+        # If an execution_timeout is set, set the timeout to the minimum of
+        # it and the trigger timeout
+        if ti.task:
+            execution_timeout = ti.task.execution_timeout
+            if execution_timeout:
+                if TYPE_CHECKING:
+                    assert ti.start_date
+                if ti.trigger_timeout:
+                    trigger_timeout = min(ti.start_date + execution_timeout, ti.trigger_timeout)
+                else:
+                    trigger_timeout = ti.start_date + execution_timeout
+
+        query = update(TI).where(TI.id == ti_id_str)
+        query = query.values(
+            state=State.DEFERRED,
+            trigger_id=trigger_row.id,
+            next_method=ti_patch_payload.next_method,
+            next_kwargs=ti_patch_payload.kwargs or {},
```

Comment:
Do we need the `or` part here? When would it be `None`?


I think Pydantic would validate that here:

https://github.com/apache/airflow/blob/4d3140dba5976acb38c3dffd4fd963514f8da634/airflow/api_fastapi/execution_api/datamodels/taskinstance.py#L81

Reply:
Generally should never be None. 
Added a safety check, let me instead define it in the datamodel itself